blog/2025-11-11-scaling-kafka-with-automq-and-tigris/index.mdx

```diff
@@ -0,0 +1,203 @@
+
+A producer writes to any AutoMQ broker. The broker streams log segments directly
+into Tigris. Tigris stores each segment as an immutable object and replicates it
+across regions. Through this design:
+
+- All brokers can access the same data without replication
+- There is no follower lag or ISR management
+- Write throughput increases because internal fan-out disappears
+
+AutoMQ retains Kafka's protocol behavior while removing the operational weight
+of traditional replication.
+
+Additionally, Tigris distributes the data globally, ensuring regional latencies
+for Producers and Consumers.
+
+![](./automq-figure-5.webp)
+
+The diagram illustrates how AutoMQ operates when backed by Tigris object
+storage. A producer sends records to any AutoMQ broker, which remains stateless
+and handles only protocol-level logic. Instead of writing log segments to local
+disk or replicating them across brokers, the broker streams each segment
+directly into Tigris. Tigris stores these segments as immutable objects per
+partition (P1, P2, P3) and replicates them across multiple regions. Because
+replication happens entirely within the storage layer, all brokers in any region
+see the same partition data without ISR management, follower lag, or
+inter-broker traffic.
+
+## Partition and cluster rebalancing
+
+### The challenges with rebalancing Kafka
+
+Kafka stores data locally on each broker. When the cluster scales in or out,
+Kafka must:
+
+- Copy partition data to new brokers
+- Rebuild replicas
+- Re-distribute leadership
+
+This involves heavy disk I/O, large data movement, and long rebalancing
+workflows. Redistributing data can take hours or days, disrupting cluster
+performance.
+
+### AutoMQ eliminates this entire burden
+
+AutoMQ eliminates this entire burden by shifting persistence to Tigris. By
+decoupling storage and compute, brokers scale independently, ensuring reads and
+writes operations are on and from Tigris, which acts as the source of truth.
+
+Because the storage layer is shared, rebalancing becomes a lightweight metadata
+update rather than a multi-gigabyte data transfer. This enables instant
+partition reassignment, automatic traffic redistribution, and a continuously
+running self-balancing engine that keeps the cluster healthy without operator
+intervention.
+
+![](./automq-figure-6.webp)
+
+## Connecting AutoMQ to Tigris
+
+Connecting AutoMQ to Tigris is as simple as adding your object storage access
+and secret keys to your environment variables:
+
+```bash
+KAFKA_S3_ACCESS_KEY=tid_YOUR_ACCESS_KEY_HERE
+KAFKA_S3_SECRET_KEY=tsec_YOUR_SECRET_KEY_HERE
+```
+
+Follow our guide on how to
+[connect Tigris to an AutoMQ instance](https://www.tigrisdata.com/docs/integrations/automq/)
+running locally using Docker.
+
+More importantly, AutoMQ's wire protocol mirrors Apache Kafka's, so existing
+clients, tooling and code remains unchanged. From an operator's standpoint,
+AutoMQ behaves like Kafka but removes the storage management dimension.
```

Comment:
yank these two paragraphs and put them under the conclusion paragraph. This is the conclusion.